lib/rules/no-console.js

```diff
@@ -36,30 +113,60 @@ module.exports = {
     },
 
     create: function(context) {
+        var globalScope;
 
         return {
+            "Program": function() {
+                globalScope = context.getScope();
+            },
 
             "MemberExpression": function(node) {
+                var currentScope = context.getScope(),
+                    passedProperty = getPropertyName(node),
+                    nestedNode = node.object,
+                    allowedProperties,
+                    propertyIsAllowed,
+                    notAllowedGlobals,
+                    globalIsNotAllowed;
 
-                if (node.object.name === "console") {
-                    var blockConsole = true;
+                if (context.options.length > 0) {
+                    if (context.options[0].allow) {
+                        allowedProperties = context.options[0].allow;
+                        propertyIsAllowed = (allowedProperties.indexOf(passedProperty) > -1);
+                    }
+                    if (context.options[0].disallowGlobals && nestedNode.object) {
+                        notAllowedGlobals = context.options[0].disallowGlobals;
+                        globalIsNotAllowed = (notAllowedGlobals.indexOf(nestedNode.object.name) > -1);
+                    }
+                }
+
+                if (nestedNode.type === "Identifier") {
 
-                    if (context.options.length > 0) {
-                        var allowedProperties = context.options[0].allow;
-                        var passedProperty = node.property.name;
-                        var propertyIsAllowed = (allowedProperties.indexOf(passedProperty) > -1);
+                    if (!isShadowed(currentScope, globalScope, node) && nestedNode.name === "console") {
```

Comment:
I guess this rule will get similar to the [radix](https://github.com/eslint/eslint/blob/32748dc8ffc9138594d8c46e4df6aa08971b5b72/lib/rules/radix.js#L126-L153) rule.